src/st/st0/2C564.c

```diff
@@ -96,86 +100,89 @@ static u32 func_801ABBBC(s32 step, Entity* dracula) {
                     prim->y1 = 171;
                 }
             } else if (prim->y1 >= 203) {
                 prim->y1 = 203;
             }
+            rgbPtr = &prim->r1;
             // Iterate through r1, g1, and b1.
-            for (redPtr = &prim->r1, var_a1 = 0; var_a1 < 3; redPtr++,
-                var_a1++) {
-                color = *redPtr;
+            for (j = 0; j < 3; j++, rgbPtr++) {
+                color = *rgbPtr;
                 if (color < 0xFF) {
                     // This ends up offsetting to r2, g2, and b2
-                    color += redPtr[0xC];
+                    color += rgbPtr[0xC];
                     if (color > 0xFF) {
                         color = 0xFF;
                     }
                 }
-                *redPtr = color;
+                *rgbPtr = color;
             }
+
             prim = prim->next;
         }
-        if (dracula->ext.dracula.unk92++ >= 25) {
+
+        if (dracula->ext.dracula.unk92++ > 24) {
             dracula->ext.dracula.unk92 = 0;
             step++;
         }
         break;
+
     case 2:
         prim = dracula->ext.dracula.unk80;
         for (i = 0; i < 16; i++) {
             prim = prim->next;
         }
+
         dracula->ext.dracula.unk84 = prim;
         for (i = 0; i < 48; i++) {
             prim->type = PRIM_LINE_G2_ALT;
             if (i == 47) {
                 prim->type = PRIM_LINE_G2;
             }
-            prim->r0 = prim->g0 = prim->b0 = 0x70;
-            prim->r1 = prim->g1 = prim->b1 = 0x20;
-            randVar2 = Random();
-            prim->r2 = (((randVar2 & 1) * 4) + 4);
-            prim->g2 = (((randVar2 * 2) & 4) + 4);
-            prim->b2 = ((randVar2 & 4) + 4);
-            if (i & 1) {
+            PGREY(prim, 0) = 0x70;
+            PGREY(prim, 1) = 0x20;
+            color = Random() & 7;
+            prim->r2 = ((color & 1) * 4) + 4;
+            prim->g2 = (((color & 2) >> 1) * 4) + 4;
+            prim->b2 = ((color >> 2) * 4) + 4;
+            if (i % 2) {
                 prim->x0 = prim->x1 = dracula->posX.i.hi + i / 2;
             } else {
                 prim->x0 = prim->x1 = dracula->posX.i.hi - i / 2;
             }
-            if (dracula->ext.dracula.unkA1 != 0) {
+            if (dracula->ext.dracula.unkA1) {
                 prim->y0 = prim->y1 = 171;
-                if (i >= 0x22) {
+                if (i > 0x21) {
                     prim->y0 = prim->y1 = -0x20;
                 }
             } else {
                 prim->y0 = prim->y1 = 203;
             }
-            prim->y1_f32.i.lo = 0;
-            prim->y1_f32.i.hi = prim->y1;
-            // FAKE variable reuse
-            color = (i >> 1);
-            prim->y1_f32_velocity.val =
-                ((color * color * color) << 8) + 0x80000;
+            F(prim->x2).i.lo = 0;
+            F(prim->x2).i.hi = prim->y1;
+            color = i >> 1;
+            F(prim->x3).val = ((color * color * color) << 8) + 0x80000;
```

Comment:
These are the only usage of anything unique to `draculaPrimitive`. Since this can be achieved with the `F()` macro on `Primitive`, I did not see much reason to keep this struct. IMO it didn't seem to be giving much additional context or clarity to this code, but I can revert this change if required.